### PR TITLE
[release/v0.1] backport multiple fixes

### PR DESCRIPTION
**Pull request by jiaqiluo**
New Issue | Original PR | Description | Supporting PR
------ | --------------- | ----------- | ------
https://github.com/rancher/rancher/issues/34453 | https://github.com/rancher/webhook/pull/45, https://github.com/rancher/webhook/pull/47 | delete the `mutatingwebhookconfigurations` when uninstalling the app |
https://github.com/rancher/rancher/issues/34320 | https://github.com/rancher/webhook/pull/46, https://github.com/rancher/webhook/pull/51 | support hardend cluster |  https://github.com/rancher/rancher/pull/34465
 
 
TODO after merging :
- [ ] cut new tag

### CODE REVIEW

**Review by StrongMonkey — APPROVED** [review]

**Review by aiyengar2 — APPROVED** [review]
LGTM

**Comment by jiaqiluo** [comment]
The backport PR can be merged after two approvals from devs.

**Pull request merged**
